core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -56,43 +59,88 @@
 public final class GrpcUtil {
 
   /**
    * {@link io.grpc.Metadata.Key} for the timeout header.
    */
-  public static final Metadata.Key<Long> TIMEOUT_KEY =
+  public static final Metadata.Key<Long> TIMEOUT_METADATA_KEY =
           Metadata.Key.of(GrpcUtil.TIMEOUT, new TimeoutMarshaller());
 
   /**
+   * @deprecated replaced by {@link #TIMEOUT_METADATA_KEY}.
+   */
+  @Deprecated
+  public static final Metadata.Key<Long> TIMEOUT_KEY = TIMEOUT_METADATA_KEY;
+
+  /**
    * {@link io.grpc.Metadata.Key} for the message encoding header.
    */
-  public static final Metadata.Key<String> MESSAGE_ENCODING_KEY =
+  public static final Metadata.Key<String> MESSAGE_ENCODING_METADATA_KEY =
           Metadata.Key.of(GrpcUtil.MESSAGE_ENCODING, Metadata.ASCII_STRING_MARSHALLER);
 
   /**
+   * @deprecated replaced by {@link #MESSAGE_ENCODING_METADATA_KEY}.
+   */
+  @Deprecated
+  public static final Metadata.Key<String> MESSAGE_ENCODING_KEY = MESSAGE_ENCODING_METADATA_KEY;
+
+  /**
    * {@link io.grpc.Metadata.Key} for the accepted message encodings header.
    */
-  public static final Metadata.Key<String> MESSAGE_ACCEPT_ENCODING_KEY =
+  public static final Metadata.Key<String> MESSAGE_ACCEPT_ENCODING_METADATA_KEY =
           Metadata.Key.of(GrpcUtil.MESSAGE_ACCEPT_ENCODING, Metadata.ASCII_STRING_MARSHALLER);
 
   /**
+   * @deprecated replaced by {@link #MESSAGE_ACCEPT_ENCODING_METADATA_KEY}.
+   */
+  @Deprecated
+  public static final Metadata.Key<String> MESSAGE_ACCEPT_ENCODING_KEY =
+          MESSAGE_ACCEPT_ENCODING_METADATA_KEY;
+
+  /**
    * {@link io.grpc.Metadata.Key} for the Content-Type request/response header.
    */
-  public static final Metadata.Key<String> CONTENT_TYPE_KEY =
+  public static final Metadata.Key<String> CONTENT_TYPE_METADATA_KEY =
           Metadata.Key.of("content-type", Metadata.ASCII_STRING_MARSHALLER);
 
   /**
+   * @deprecated replaced by {@link #CONTENT_TYPE_METADATA_KEY}.
+   */
+  @Deprecated
+  public static final Metadata.Key<String> CONTENT_TYPE_KEY = CONTENT_TYPE_METADATA_KEY;
+
+  /**
    * {@link io.grpc.Metadata.Key} for the Content-Type request/response header.
    */
-  public static final Metadata.Key<String> USER_AGENT_KEY =
+  public static final Metadata.Key<String> USER_AGENT_METADATA_KEY =
           Metadata.Key.of("user-agent", Metadata.ASCII_STRING_MARSHALLER);
 
   /**
+   * @deprecated replaced by {@link #USER_AGENT_METADATA_KEY}.
+   */
+  @Deprecated
+  public static final Metadata.Key<String> USER_AGENT_KEY = USER_AGENT_METADATA_KEY;
+
+  /**
+   * {@link io.grpc.Attributes.Key} for the remote address of stream call.
+   */
+  public static final Attributes.Key<SocketAddress> REMOTE_ADDR_CALL_ATTR_KEY =
+          Attributes.Key.of("io.grpc.RemoteAddr");
+
+  /**
+   * {@link io.grpc.Attributes.Key} for the SSL session of stream call.
+   */
+  public static final Attributes.Key<Optional<SSLSession>> SSL_SESSION_CALL_ATTR_KEY =
```

Comment:
That's fine. We have a bit of aversion to Optional because of the mess caused by Guava's vs JDK's Optional with no great migration path. If we could use JDK's Optional, then we probably wouldn't have much issue with using Optional.